mm/2s2h/Enhancements/Minigames/Archery.cpp

```diff
@@ -51,4 +55,10 @@ void RegisterArchery() {
     });
 }
 
-static RegisterShipInitFunc initFunc(RegisterArchery, { SWAMP_CVAR_NAME, TOWN_CVAR_NAME });
+static void RegisterBoatArchery() {
+    COND_HOOK(AfterEndOfCycleSave, BOAT_CVAR != 20, []() { HS_SET_BOAT_ARCHERY_HIGH_SCORE(BOAT_CVAR - 1); });
+}
```

Comment:
`AfterEndOfCycleSave` means that this will not apply until the player SoT saves after configuring this setting, or at all in the first cycle after creating a save (e.g. in rando or with first cycle skip). Since this runs within ShipInit and thus executes any time the setting is changed, you can probably just:

```suggestion
static void RegisterBoatArchery() { HS_SET_BOAT_ARCHERY_HIGH_SCORE(BOAT_CVAR - 1); }
```

Reply:
This actually needs a major rethink if we want the score to update live like that. First of all, TIL that Koume tells you that "your best score" is 19 hits even though you never hit that mark. Second, this needs to be compatible with the Swamp Boat Timesaver hook. Third, this isn't just a "perfect score" like the other archery minigames - it's a running high score that requires preservation. Fourth, it's possible to _increase_ the target value, in which case just comparing it to that setting isn't going to work.